src/status_im/utils/transducers.cljs

```diff
@@ -0,0 +1,26 @@
+(ns status-im.utils.transducers
+  "Utility namespace containing various usefull transducers")
+
+(defn last-distinct-by
+  "Just like regular `distinct`, but you provide function
+  computing the distinctness of input elements and when
+  duplicate elements are removed, the last, not the first
+  one is removed."
+  [compare-fn]
+  (fn [rf]
+    (let [accumulated-input (volatile! {:seen {}
```

Comment:
Why do we use `volatile!`? I'd rather avoid this unless we have benchmarks that prove it is necessary for perf.

Reply:
Because it's a standard reference type for encapsulated state inside transducers which is not shared among threads (which is not even relevant for js env), see all the standard stateful transducers from `clojure.core`, for example `dedupe` - https://github.com/clojure/clojure/blob/clojure-1.9.0-alpha14/src/clj/clojure/core.clj#L7534-#L7545